tower-posix-tests/tower-posix-tests.cabal

```diff
@@ -1,4 +1,4 @@
-name:                ivory-tower-posix
+name:                tower-posix-tests
```

Comment:
Rather than splitting this off into a separate package, let's make the test executables into `test-suite` sections like in https://github.com/GaloisInc/smaccmpilot-stm32f4/blob/master/src/ivory-px4-hw/ivory-px4-hw.cabal